x-pack/test/security_solution_endpoint_api_int/apis/resolver/entity_id.ts

```diff
@@ -0,0 +1,156 @@
+        );
+        expect(body).to.be.empty();
+      });
+    });
+
+    describe('children', () => {
+      let origin: Event;
+      let childNoEntityID: Event;
+      let childWithEntityID: Event;
+      let events: Event[];
+      let genData: InsertedEvents;
+
+      before(async () => {
+        // construct a tree with an origin and two direct children. One child will not have an entity_id. That child
+        // should not be returned by the backend.
+        origin = generator.generateEvent({ entityID: 'a' });
+        childNoEntityID = generator.generateEvent({
+          parentEntityID: origin.process.entity_id,
+          ancestry: [origin.process.entity_id],
+        });
+        // force it to be empty
+        childNoEntityID.process.entity_id = '';
+
+        childWithEntityID = generator.generateEvent({
+          entityID: 'b',
+          parentEntityID: origin.process.entity_id,
+          ancestry: [origin.process.entity_id],
+        });
+        events = [origin, childNoEntityID, childWithEntityID];
+        genData = await resolver.insertEvents(events);
+      });
+
+      after(async () => {
+        await resolver.deleteData(genData);
+      });
+
+      it('does not find children without a process entity_id', async () => {
+        const { body }: { body: ResolverTree } = await supertest
+          .get(`/api/endpoint/resolver/${origin.process.entity_id}`)
+          .expect(200);
+        expect(body.children.childNodes.length).to.be(1);
+        expect(body.children.childNodes[0].entityID).to.be(childWithEntityID.process.entity_id);
+      });
+    });
+
+    describe('ancestors', () => {
+      let origin: Event;
+      let ancestor1: Event;
+      let ancestor2: Event;
+      let ancestorNoEntityID: Event;
+      let events: Event[];
+      let genData: InsertedEvents;
+
+      before(async () => {
+        // construct a tree with an origin that has two ancestors. The origin will have an empty string as one of the
+        // entity_ids in the ancestry array. This is to make sure that the backend will not query for that event.
+        ancestor2 = generator.generateEvent({
+          entityID: '2',
+        });
+        ancestor1 = generator.generateEvent({
+          entityID: '1',
+          parentEntityID: ancestor2.process.entity_id,
+          ancestry: [ancestor2.process.entity_id],
+        });
+
+        // we'll insert an event that doesn't have an entity id so if the backend does search for it, it should be
+        // returned and our test should fail
+        ancestorNoEntityID = generator.generateEvent({
+          ancestry: [ancestor2.process.entity_id],
+        });
+        ancestorNoEntityID.process.entity_id = '';
+
+        origin = generator.generateEvent({
+          entityID: 'a',
+          parentEntityID: ancestor1.process.entity_id,
+          ancestry: ['', ancestor2.process.entity_id],
```

Comment:
you could just make the first one here `ancestorNoEntityID.process.entity_id` too. Took me a sec to figure that's what was being referenced. Nothing to restart CI over though

Reply:
Yeah good point. I added that late. I wanted to make sure that the backend wouldn't return the event even if ES actually did have a document with entity_id as an empty string